crates/sui-core/src/consensus_adapter.rs

```diff
@@ -235,6 +299,11 @@ impl ConsensusAdapter {
             .epoch_tables()
             .pending_consensus_transactions
             .insert(&transaction.key(), &transaction)?;
+        if let ConsensusTransactionKind::UserTransaction(certificate) = &transaction.kind {
+            self.pending_certificates
+                .lock()
```

Comment:
Note that here we are locking pending_certificates without holding a lock to reconfig state

Reply:
We actually do, holding it in `_lock` variable